src/SwaggerProvider.DesignTime/v2/Parser/Parsers.fs

```diff
@@ -401,18 +401,69 @@ module Parsers =
+        {
+            Title = obj.GetRequiredField("title", spec).AsString()
+            Description = obj.GetStringSafe("description")
+            Version = obj.GetRequiredField("version", spec).AsString()
+        }
+
+    /// Parses the SchemaNode as a TagObject.
+    let parseTagObject (obj:SchemaNode) : TagObject =
+        let spec = "http://swagger.io/specification/#tagObject"
+        {
+            Name = obj.GetRequiredField("name", spec).AsString()
+            Description = obj.GetStringSafe("description")
+        }
+
     /// Parse the SchemaNode as a PathItemObject[]
-    let parsePathsObject (context:ParserContext) (obj:SchemaNode) : OperationObject[] =
+    let rec parsePathsObject (resolveReferences:bool) (parser:string->SchemaNode) (context:ParserContext) (obj:SchemaNode) : (OperationObject option*SwaggerObject option)[] =
         let parsePathItemObject (context:ParserContext) path (field, obj) =
             match field with
-            | "get"     -> Some <| parseOperationObject context path Get obj
-            | "put"     -> Some <| parseOperationObject context path Put obj
-            | "post"    -> Some <| parseOperationObject context path Post obj
-            | "delete"  -> Some <| parseOperationObject context path Delete obj
-            | "options" -> Some <| parseOperationObject context path Options obj
-            | "head"    -> Some <| parseOperationObject context path Head obj
-            | "patch"   -> Some <| parseOperationObject context path Patch obj
-            | "$ref"    -> failwith "External definition of this path item is not supported yet"
+            | "get"     -> Some <| (Some <| parseOperationObject context path Get obj, None)
+            | "put"     -> Some <| (Some <| parseOperationObject context path Put obj, None)
+            | "post"    -> Some <| (Some <| parseOperationObject context path Post obj, None)
+            | "delete"  -> Some <| (Some <| parseOperationObject context path Delete obj, None)
+            | "options" -> Some <| (Some <| parseOperationObject context path Options obj, None)
+            | "head"    -> Some <| (Some <| parseOperationObject context path Head obj, None)
+            | "patch"   -> Some <| (Some <| parseOperationObject context path Patch obj, None)
+            | "$ref"    ->
+                if not resolveReferences then
+                    failwith "SwaggerProvider static parameter ResolveReferences has to be set true to parse External definition of this path item"
+                else
+                let fileName =
+                    let fname = obj.AsString()
+                    (if fname.Contains("#") then fname.Split('#').[0] else fname)
+                let execAssembly =
+#if INTERACTIVE
+                    "" //not supported
+#else
+                    System.Reflection.Assembly.GetExecutingAssembly().Location
+#endif
+                let tryPaths =
+                    [ fileName;
```

Comment:
Why do not you use `Schema` TP parameter (url or file path) without filename as an option (1st option) as base.
For example if we  build TP for `http://my.server.com/v2/swagger.yaml` and found `../v2/token.yaml#/token` it most likely refer to type from `http://my.server.com/v2/token.yaml`